# Análisis sobre las expectativas de las mujeres al salir de prisión en México
#### Se examinan datos provenientes de la Encuesta Nacional de Población Privada de la Libertad (ENPOL 2021) la cual realizó el Instituto Nacional de Estadadística y Geografía de México.

<details>
    <summary><strong>Meta</strong></summary>
    En este notebook se muestra el proceso que se ha seguido para responder a la pregunta: ¿Cómo consideran que será su vida en libertad las mujeres que se encuentran en prisión en México? 
    Para responder a esta cuestión es necesario realizar un Análisis Exploratorio de Datos, con el fin de conocer las variables más adecuadas para responder a la pregunta planteada, hacer una descripción estadística de las mismas y proceder a una visualización que permita ilustrar los resultados. 
    <ul>
        Metas específicas </li>
        <li> 1: Identificar variables de interes en los tres conjuntos de datos que se escogieron con antelación y que llevan por nombre ENPOL2021_SOC, ENPOL2021_5, ENPOL2021_8_9_10_11</strong></i> </li>
        <li> 2: Crear un Dataset base que contenga las variables adecuadas extraidas de los datos en bruto.</li>
        <li> 3: Realizar un Análisis Exploratorio y Visualización de los datos.</li>
    </ul>
</details>

<details>
    <summary><strong>Contexto</strong></summary>
    Los archivos en formato .csv fueron obtenidos del portal oficial del Instituto Nacional de Geografía y Estadística. Es importante resaltar que para este proyecto solo se han tomado tres archivos que corresponden a secciones especificas de la Encuesta mencionada, las cuales tratan sobre: Características Sociodemográficas (ENPOL2021_SOC), Proceso Judicial (ENPOL2021_5) y Expectativas de Salida (ENPOL2021_8_9_10_11). </li>

El primer conjunto de datos permite conocer el perfil de la persona en prisión a través de su edad, estado civil, entorno familiar, pertenecia étnica, escolaridad, identidad, orientación sexual, estado de salud, entre otros aspectos. El segundo grupo está dividido entre personas sentenciadas y procesadas y se incluyen preguntas vinculadas al proceso judical que han vivido. Finalmente el tercer agregado considera apectos como: la corrupción al interior del centro penitenciario, antecedentes juridico-penales y las expectativas de salida de las personas en prisión. Como se podrá ver, es primordial filtrar las variables para obtener solo las vinculadas con nuestro interés principal y añadirlas en una unidad de análisis, con ello será más sencillo explorar los datos y visualizar los resultados.
<ul>
</details>

**Paso 0:** Carga de las bibliotecas básicas para leer los datos brutos. 

In [1]:
import numpy as np
import pandas as pd
import altair as alt
from pathlib import Path
from herramientas import arbol

La biblioteca **pathlib** permite navegar en las rutas donde se encuentran los datos del proyecto y que éstas se puedan leer en diferentes sistemas operativos. En este caso, para la estructura del proyecto, la lectura de las rutas y el uso de la herramienta "árbol" se retomó la [plantilla](https://github.com/tacosdedatos/cookiecutter-analisis-de-datos) creada en cookiecutter por Sergio Sánchez.

In [2]:
DATOS_BRUTOS = Path("../datos/brutos/")
DATOS_INTERINOS = Path("../datos/interinos/")
DATOS_PROCESADOS = Path("../datos/procesados/")
DATOS_EXTERNOS = Path("../datos/externos/")

In [3]:
arbol(DATOS_BRUTOS)

+ ..\datos\brutos
    + ENPOL2021_5.csv
    + ENPOL2021_8_9_10_11.csv
    + ENPOL2021_SOC.csv


Por cada archivo .csv se creo un DataFrame, para identificarlos se usaron nombres de variables parecidos a los nombres del archivo original. La lectura de los archivos fue más secilla ya que con el paso anterior cada ruta fue guardada en una variable.  

En la lectura del archivo correspondiente a la variable `enpol_soc` hubo un problema de reconocimiento con la advertencia "Solve DtypeWarning: Columns (5) have mixed types". Para solucionarlo se opto por especificar el dtype de la columna 5, esta solución se adoptó después de consultar el sitio de [Roel Peters](https://www.roelpeters.be/solved-dtypewarning-columns-have-mixed-types-specify-dtype-option-on-import-or-set-low-memory-in-pandas/).

In [4]:
enpol_soc = pd.read_csv(DATOS_BRUTOS / "ENPOL2021_SOC.csv", dtype = {"NOM_INT": "object"}, encoding = "ISO-8859-1") 

In [5]:
enpol_5 = pd.read_csv(DATOS_BRUTOS / "ENPOL2021_5.csv", encoding = "ISO-8859-1")

In [6]:
enpol_10 = pd.read_csv(DATOS_BRUTOS / "ENPOL2021_8_9_10_11.csv", encoding = "ISO-8859-1")

Una vez que los archivos han sido cargados para su lectura como DataFrames es importante dar una primera mirada general para conocer su tamaño, las columnas por las que están compuestos, su orden y el tipo de datos que continen. Esto dará la pauta para saber qué cambios se deben realizar para obtener un DataFrame que contenga solo los elementos que necesitamos. 

In [7]:
enpol_soc

Unnamed: 0     ID_PER  CVE_ENT         NOM_ENT  CEN_INT  \
0               1  2173.8228        1  Aguascalientes      1.0   
1               2  1791.0884        1  Aguascalientes      1.0   
2               3   188.2951        1  Aguascalientes      1.0   
3               4  1510.0944        1  Aguascalientes      1.0   
4               5  1784.9143        1  Aguascalientes      1.0   
...           ...        ...      ...             ...      ...   
61444       81184  2895.6017       32       Zacatecas      NaN   
61445       81185  1555.7952       32       Zacatecas      NaN   
61446       81187  1408.5145       32       Zacatecas      NaN   
61447       81188   570.8920       32       Zacatecas      NaN   
61448       81189  1528.4766       32       Zacatecas      NaN   

                     NOM_INT  COD_RES  SEXO  FUERO  P1_1  ...  P1_42_14  \
0      CERESO AGUASCALIENTES        1     1      2     1  ...       NaN   
1      CERESO AGUASCALIENTES        1     1      2     1  ...       NaN   
2      CERESO AGUASCALIENTES        1     1      2     1  ...       NaN   
3      CERESO AGUASCALIENTES        1     1      2     1  ...       NaN   
4      CERESO AGUASCALIENTES        1     1      2     1  ...       NaN   
...                      ...      ...   ...    ...   ...  ...       ...   
61444                    NaN        1     2      2     2  ...       NaN   
61445                    NaN        1     2      2     2  ...       NaN   
61446                    NaN        1     2      2     2  ...       NaN   
61447                    NaN        1     2      2     2  ...       NaN   
61448                    NaN        1     2      2     2  ...       NaN   

       P1_43_14  P1_44_14  P1_45  P1_46  P1_47  P1_48  EST_DIS   FAC_PER  \
0           NaN       NaN    NaN    NaN    1.0      4   3029.1  2.473585   
1           NaN       NaN    2.0    NaN    2.0      4   3029.1  2.473585   
2           NaN       NaN    NaN    NaN    NaN      4   3029.1  2.473585   
3           NaN       NaN    1.0    2.0    NaN      4   3029.1  2.473585   
4           NaN       NaN    2.0    NaN    2.0      4   3029.1  2.473585   
...         ...       ...    ...    ...    ...    ...      ...       ...   
61444       NaN       NaN    NaN    NaN    NaN      4   1128.2  1.062500   
61445       NaN       NaN    2.0    NaN    1.0      1   1128.2  1.062500   
61446       NaN       NaN    NaN    NaN    NaN      4   1128.2  1.062500   
61447       NaN       NaN    NaN    NaN    NaN      4   1128.2  1.062500   
61448       NaN       NaN    NaN    NaN    2.0      4   1128.2  1.062500   

            FPC  
0      0.438017  
1      0.438017  
2      0.438017  
3      0.438017  
4      0.438017  
...         ...  
61444  1.000000  
61445  1.000000  
61446  1.000000  
61447  1.000000  
61448  1.000000  

[61449 rows x 182 columns]

In [8]:
enpol_5

Unnamed: 0     ID_PER  CVE_ENT           NOM_ENT  CEN_INT  \
0               1   101.0001       13           Hidalgo      NaN   
1               2   101.1432        1    Aguascalientes      NaN   
2               3   101.1477        9  Ciudad de México     15.0   
3               4   101.2131       32         Zacatecas      NaN   
4               5   101.2980       28        Tamaulipas      NaN   
...           ...        ...      ...               ...      ...   
61444       61445  3203.6415        7           Chiapas      NaN   
61445       61446  3203.7197       24   San Luis Potosí     46.0   
61446       61447  3203.7425       31           Yucatán     53.0   
61447       61448  3203.8400       23      Quintana Roo      NaN   
61448       61449  3204.0276       13           Hidalgo      NaN   

                                     NOM_INT  COD_RES  SEXO  FUERO  P5_1  ...  \
0                                        NaN        1     1      2     1  ...   
1                                        NaN        1     2      2     1  ...   
2      RECLUSORIO PREVENTIVO VARONIL ORIENTE        1     1      2     2  ...   
3                                        NaN        1     1      2     1  ...   
4                                        NaN        1     1      2     2  ...   
...                                      ...      ...   ...    ...   ...  ...   
61444                                    NaN        1     1      2     1  ...   
61445          PENITENCIARIA SAN LUIS POTOSI        1     1      2     1  ...   
61446                          CERESO MERIDA        1     1      2     1  ...   
61447                                    NaN        1     1      2     2  ...   
61448                                    NaN        1     1      2     1  ...   

       P5_46_2  P5_46_3  P5_46_4  P5_46_5  P5_46_6  P5_47   P5_48  EST_DIS  \
0          NaN      NaN      NaN      NaN      NaN    NaN     NaN    883.1   
1          NaN      NaN      NaN      NaN      NaN    NaN     NaN    872.2   
2          1.0      2.0      2.0      2.0      1.0    1.0  5000.0    210.1   
3          NaN      NaN      NaN      NaN      NaN    NaN     NaN   2294.1   
4          NaN      NaN      NaN      NaN      NaN    NaN     NaN   1979.1   
...        ...      ...      ...      ...      ...    ...     ...      ...   
61444      2.0      2.0      2.0      2.0      2.0    2.0     NaN   2172.1   
61445      NaN      NaN      NaN      NaN      NaN    NaN     NaN    163.1   
61446      NaN      NaN      NaN      NaN      NaN    NaN     NaN   2884.1   
61447      NaN      NaN      NaN      NaN      NaN    NaN     NaN   2351.1   
61448      NaN      NaN      NaN      NaN      NaN    NaN     NaN    177.1   

         FAC_PER       FPC  
0       3.981132  0.258537  
1       1.050000  0.991736  
2      11.704167  0.198020  
3       3.228571  0.416667  
4       3.045977  0.368644  
...          ...       ...  
61444   4.746667  0.222552  
61445   2.588732  0.373291  
61446   2.346723  0.340778  
61447   2.888199  0.334025  
61448   4.238095  0.259259  

[61449 rows x 241 columns]

In [9]:
enpol_10

Unnamed: 0     ID_PER  CVE_ENT           NOM_ENT  CEN_INT  \
0               1   101.0001       13           Hidalgo      NaN   
1               2   101.1432        1    Aguascalientes      NaN   
2               3   101.1477        9  Ciudad de México     15.0   
3               4   101.2131       32         Zacatecas      NaN   
4               5   101.2980       28        Tamaulipas      NaN   
...           ...        ...      ...               ...      ...   
61444       61445  3203.6415        7           Chiapas      NaN   
61445       61446  3203.7197       24   San Luis Potosí     46.0   
61446       61447  3203.7425       31           Yucatán     53.0   
61447       61448  3203.8400       23      Quintana Roo      NaN   
61448       61449  3204.0276       13           Hidalgo      NaN   

                                     NOM_INT  COD_RES  SEXO  FUERO  P8_1_01  \
0                                        NaN        1     1      2        0   
1                                        NaN        1     2      2        0   
2      RECLUSORIO PREVENTIVO VARONIL ORIENTE        1     1      2        1   
3                                        NaN        1     1      2        0   
4                                        NaN        1     1      2        0   
...                                      ...      ...   ...    ...      ...   
61444                                    NaN        1     1      2        0   
61445          PENITENCIARIA SAN LUIS POTOSI        1     1      2        0   
61446                          CERESO MERIDA        1     1      2        0   
61447                                    NaN        1     1      2        0   
61448                                    NaN        1     1      2        0   

       ...  P10_11  P11_1_1  P11_1_2  P11_1_3  P11_1_4  P11_1_8  P11_1_9  \
0      ...       F        0        0        0        1        0        0   
1      ...       H        0        0        0        1        0        0   
2      ...       G        0        0        0        1        0        0   
3      ...       H        0        0        0        1        0        0   
4      ...       G        0        0        0        1        0        0   
...    ...     ...      ...      ...      ...      ...      ...      ...   
61444  ...       G        0        0        1        0        0        0   
61445  ...       H        0        0        0        1        0        0   
61446  ...       H        0        1        0        0        0        0   
61447  ...       H        0        0        0        1        0        0   
61448  ...       F        0        0        0        1        0        0   

       EST_DIS    FAC_PER       FPC  
0        883.1   3.981132  0.258537  
1        872.2   1.050000  0.991736  
2        210.1  11.704167  0.198020  
3       2294.1   3.228571  0.416667  
4       1979.1   3.045977  0.368644  
...        ...        ...       ...  
61444   2172.1   4.746667  0.222552  
61445    163.1   2.588732  0.373291  
61446   2884.1   2.346723  0.340778  
61447   2351.1   2.888199  0.334025  
61448    177.1   4.238095  0.259259  

[61449 rows x 157 columns]

In [10]:
#enpol_soc.info(object)
#enpol_5.info(object)
#enpol_10.info(object

## Observaciones

1. El número de filas coincide en cada Dataframe (61,449). Esto es una forma de verificar que tenemos el mismo número de personas que respondieron la encuesta. 
2. Naturalmente el número de filas difiere ya que cada dataframe está compuesto por diferentes secciones de la encuesta y cada sección está formada por un número de preguntas específico. 
3. Tanto en **enpol_soc**, como en **enpol_5** y **enpol_10**, las primeras 9 columnas tiene el mismo nombre y esto se debe a que se refieren a los Datos de indentificación de las personas entrevistadas. 
4. El significado del nombre de las columnas se encuentra en el [diccionario](https://www.inegi.org.mx/contenidos/programas/enpol/2021/microdatos/fd_enpol2021.pdf) creado por el Instituto, sobre las 9 variables mencionadas tenemos que se refieren a: 
* **Unnamed** Número consecutivo de cada registro 
* **ID_PER** Identificador de persona
* **CVE_ENT** Entidad (Númerico)
* **NOM_ENT** Nombre de la Entidad
* **CEN_INT** Clave del Centro de Interés 
* **NOM_INT** Nombre del Centro de Interés
* **COD_RES** Entrevista completa 
* **SEXO** Hombre / Mujer
* **FUERO** Centro Federal / Centro Estatal o Municipal
5. Al prestar atención a la columna **ID_PER** del dataframe **enpol_soc** se puede ver que tiene un orden distinto respecto a **enpol_5** y **enpol_10**, ya que los datos del primer conjunto están ordenados alfabeticamente con base en el nombre de la Entidad (**NOM_ENT**).
6. Por último, tenemos que gracias a `.info(object)` se puede acceder al nombre de la columna, su índice y el dtype que contiene. Nuevamente para entender a qué pregunta hace referencia cada clave se cuenta con el [diccionario](https://www.inegi.org.mx/contenidos/programas/enpol/2021/microdatos/fd_enpol2021.pdf) mencionado. Así mismo, para identificar las secciones que contine cada archivo se puede consultar el [cuestionario](https://www.inegi.org.mx/contenidos/programas/enpol/2021/doc/2021_enpol_cuestionario_principal.pdf) como fue aplicado, la lectura de este documento es importante porque permite observar el orden que siguió el entrevistador al aplicar la encuesta. 


## Creación del DataFrame principal 

Con base en las observaciones anteriores tenemos la pauta para continuar con la creación un DataFrame que permita hacer el análisis sobre las expectativas de las mujeres al salir de prisión. 

* A **enpol_soc** se le ha agregado la función `.sort_values` para ordenar la información tomando la variable "Identificador de persona".
* Tanto en **enpol_soc** como en **enpol_5** y **enpol_10** se extrajeron las variables de nuestro interés, las cuales se escogieron al hacer una lectura previa del [diccionario](https://www.inegi.org.mx/contenidos/programas/enpol/2021/microdatos/fd_enpol2021.pdf) y [cuestionario](https://www.inegi.org.mx/contenidos/programas/enpol/2021/doc/2021_enpol_cuestionario_principal.pdf). Así mismo se cambio el indice de cada conjunto para que con esta columna coincidieran los registros. 
* Con `pd.concat` se unieron los datos de los tres conjuntos separados.
* Finalmente, del DataFrame nombrado **enpol_expectativas** se filtro la información por SEXO == 2 [MUJER]

In [11]:
enpol_soc_ordenado_por_identificador = enpol_soc.sort_values(by = "ID_PER")

In [12]:
enpol_soc_expectativas = enpol_soc_ordenado_por_identificador[["ID_PER", "NOM_ENT", "NOM_INT", "SEXO", "FUERO", "P1_1", "P1_1A", "P1_3", "P1_7", "P1_8", "P1_9", "P1_11_1", "P1_11_2", "P1_11_3", "P1_11_4", "P1_11_5", "P1_11_6", "P1_11_7", "P1_11_8", "P1_11_9", "P1_15", "P1_18_N", "P1_18_G", "P1_22", "P1_23"]].set_index("ID_PER")

In [13]:
enpol_5_expectativas = enpol_5[["ID_PER","P5_4_A", "P5_4_M", "P5_10", "P5_28", "P5_34_A", "P5_34_M"]].set_index("ID_PER")

In [14]:
enpol_10_expectativas = enpol_10[["ID_PER", "P10_1", "P10_2", "P10_3", "P10_4_1", "P10_4_2", "P10_4_3", "P10_4_4", "P10_4_5", "P10_4_6", "P10_4_8", "P10_4_9", "P10_5_1", "P10_5_2", "P10_5_3", "P10_5_4", "P10_6", "P10_7", "P10_8", "P10_9", "P10_10", "P10_11"]].set_index("ID_PER")

In [15]:
enpol_expectativas = pd.concat([enpol_soc_expectativas, enpol_5_expectativas, enpol_10_expectativas], axis  = 1)

In [16]:
enpol_expectativas_mujeres_1 = enpol_expectativas[enpol_expectativas["SEXO"] == 2]

In [17]:
enpol_expectativas_mujeres = enpol_expectativas_mujeres_1.drop(columns = "SEXO")

In [18]:
enpol_expectativas_mujeres.reset_index(inplace = True, drop = False)

In [19]:
enpol_expectativas_mujeres

ID_PER                          NOM_ENT  \
0       101.1432                   Aguascalientes   
1       101.3717                           Puebla   
2       101.3859              Michoacán de Ocampo   
3       101.5967  Veracruz de Ignacio de la Llave   
4       101.7186                          Jalisco   
...          ...                              ...   
11701  3201.6414                 Ciudad de México   
11702  3202.1412              Michoacán de Ocampo   
11703  3202.5767                        Chihuahua   
11704  3203.2904                          Morelos   
11705  3203.3336                         Guerrero   

                                                 NOM_INT  FUERO  P1_1  P1_1A  \
0                                                    NaN      2     2      5   
1                                                    NaN      2     3      1   
2                                                    NaN      2     1      5   
3                                                    NaN      2     3      5   
4                       COMISARIA DE REINSERCION FEMENIL      2     2      5   
...                                                  ...    ...   ...    ...   
11701  CENTRO FEMENIL DE REINSERCIÓN SOCIAL SANTA MAR...      2     2      2   
11702                                                NaN      2     2      5   
11703                                                NaN      2     2      5   
11704                            CEFERESO 16 CPS MORELOS      1     2      2   
11705                                                NaN      2     3      5   

       P1_3  P1_7  P1_8  P1_9  ...  P10_5_1  P10_5_2  P10_5_3  P10_5_4  P10_6  \
0        40     4     1   4.0  ...        2        2        2        2      1   
1        32     1     1   4.0  ...        1        2        1        1      2   
2        30     4     1   2.0  ...        2        2        1        2      1   
3        53     1     2   NaN  ...        1        3        2        2      1   
4        28     4     1   1.0  ...        1        1        1        1      1   
...     ...   ...   ...   ...  ...      ...      ...      ...      ...    ...   
11701    51     5     2   NaN  ...        1        2        2        2      2   
11702    41     1     1   1.0  ...        2        2        2        2      1   
11703    27     1     1   2.0  ...        2        2        1        2      1   
11704    26     1     1   2.0  ...        1        2        2        2      2   
11705    22     5     2   NaN  ...        1        1        1        1      1   

       P10_7  P10_8  P10_9  P10_10  P10_11  
0          4      1    1.0       H       H  
1          4      1    1.0       J       H  
2          4      1    1.0       H       H  
3          4      2    NaN       H       H  
4          4      2    NaN       G       H  
...      ...    ...    ...     ...     ...  
11701      4      2    NaN       H       H  
11702      4      1    1.0       H       H  
11703      4      2    NaN       H       H  
11704      4      2    NaN       H       G  
11705      4      2    NaN       H       G  

[11706 rows x 51 columns]

## Descripción del proceso general que se ha seguido para la Exploración y Visualización de los datos 
* El análisis y visualización se ha dividido en tres secciones tituladas: "Características sociodemográficas de interés", "Proceso judicial" y "Expectativas de salida", éstas corresponden las orden dado en la encuesta original. 
* Cada sección está compuesta por subsecciones en las que se trata generalmente una o más de las variables escogidas. 
* Cada subsección con la descripción de lo que trata se encuentra en una celda de tipo Markdown. En una línea se específica de que trata la subsección, también se enumeran las columnas que se han extraido del dataset "enpol_expectativas_mujeres" y finalmente en una toolist se señalan los valores de las respuestas con el significado que guardan, solo los de las columnas que corresponden a preguntas. 
* En relación con el código se ha seguido el proceso siguiente: 
1. Por cada subsección se creó un nuevo dataset que extrae las variables del conjunto de datos "enpol_expectativas_mujeres", en algunos casos fue necesario filtrar los valores nulos. 
2. Se crearon mascaras para filtrar respuestas no válidas ya que éstas pueden alterar las estadísticas de los datos. Las respuestas que se han considerado como no validas generalmente corresponden a los códigos: 8) No sabe, 9) No responde, 98) No sabe, 99) No responde.
3. Al filtrar el dataset de las respuestas no válidas se procede a una descripción con `.describe`
4. Los datos se agrupan y se cuentan de acuerdo con la variable que se quiere examinar, esto gracias a `.groupby()` y `.count()`
5. Debido a que las respuestas se guardaron en números que corresponden a una variable de tipo categórico se procede a añadir una columna que contengan el significado de esas observaciones. 
6. Los resultados de la variable se guardan en un gráfico. 
7. En algunos casos fue necesario añadir un paso extra, ya que la misma pregunta fue guardada en diferentes columnas debido a que la posible respuesta era lo único que cambiaba, en este caso fue necesario filtrar en cada columna la respuesta positiva y posteriomente concatenarlas en un nuevo dataframe para así hacer posible la visualización de los resultados, este fue el caso en las preguntas: "Actualmente, ¿quién(es) se hace(n) responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años?", "¿Por qué considera que no tendría a dónde llegar a vivir?", "Pensando en el momento en el que usted esté en libertad ¿considera que el haber estado en un Centro penitenciario afecte sus posibilidades de...?"

## Características sociodemográficas de interés

Tiempo que llevan en prisión las personas que se identificaron con sexo femenino 
1. Persona == **ID_PER**
2. Pregunta ¿Cuánto tiempo tiene privado de su libertad? == **P1_1A**
* **1** == Menos de 6 meses 
* **2** == Más de 6 meses hasta 1 año 
* **3** == Más de 1 año hasta 1 año con 6 meses 
* **4** == Más de 1 año con 6 meses hasta 2 años 
* **5** == Más de 2 años 
* **8** == No sabe 
* **9** == No responde 

In [20]:
tiempo_en_prision = enpol_expectativas_mujeres[["ID_PER", "P1_1A"]]

In [21]:
tiempo_en_prision_no_sabe = tiempo_en_prision["P1_1A"] == 8

In [22]:
tiempo_en_prision_no_responde = tiempo_en_prision["P1_1A"] == 9

In [23]:
tiempo_en_prision [~ tiempo_en_prision_no_sabe & ~ tiempo_en_prision_no_responde]["P1_1A"].describe()

count    11689.000000
mean         3.779194
std          1.589157
min          1.000000
25%          2.000000
50%          5.000000
75%          5.000000
max          5.000000
Name: P1_1A, dtype: float64

In [24]:
tiempo_en_prision = tiempo_en_prision [~ tiempo_en_prision_no_sabe & ~ tiempo_en_prision_no_responde].groupby(["P1_1A"]).count()

In [25]:
tiempo = ["Menos de 6 meses", "Mas de 6 meses a 1 año", "Mas de 1 año a 1 año 6 meses", "Mas de 1 año 6 ms a 2 años", "Hasta mas de 2 años"]

In [26]:
tiempo_en_prision["Tiempo privadas de libertad"] = tiempo

In [27]:
primera_columna = tiempo_en_prision.pop("Tiempo privadas de libertad")
tiempo_en_prision.insert(0, "Tiempo privadas de libertad", primera_columna)
tiempo_en_prision

Tiempo privadas de libertad  ID_PER
P1_1A                                      
1                  Menos de 6 meses    1918
2            Mas de 6 meses a 1 año    1347
3      Mas de 1 año a 1 año 6 meses     785
4        Mas de 1 año 6 ms a 2 años     987
5               Hasta mas de 2 años    6652

In [28]:
grafico_tiempo_en_prision = alt.Chart(tiempo_en_prision).mark_bar(color = "#8267BE").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "), 
    alt.Y("Tiempo privadas de libertad:O", title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Numero de informantes")],
).properties(
    title = "Tiempo que llevan en prisión las informantes identificadas con sexo femenino",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(
    width=700, height=150
).configure_title( 
  offset = 15, fontSize = 15,
)

In [29]:
grafico_tiempo_en_prision

alt.Chart(...)

Edad de las personas que se identificaron con sexo femenino
1. Persona == **ID_PER**
2. Pregunta ¿Cuántos años cumplidos tiene? == **P1_3**
   * **18...96** == Edad
   * **97** == 97 años o más
   * **98** == No sabe 
   * **99** == No responde 

In [30]:
edad = enpol_expectativas_mujeres[["ID_PER", "P1_3"]]

In [31]:
noventa_y_siete_mas = edad["P1_3"] == 97

In [32]:
edad_no_sabe = edad["P1_3"] == 98

In [33]:
edad_no_responde = edad["P1_3"] == 99 

In [34]:
edad[~ edad_no_sabe & ~ edad_no_responde]["P1_3"].describe()

count    11699.000000
mean        35.249338
std         10.196294
min         18.000000
25%         27.000000
50%         33.000000
75%         42.000000
max         84.000000
Name: P1_3, dtype: float64

In [35]:
edad["edad_agrupada"] = pd.cut(
    edad["P1_3"], 
    bins = [18, 24, 29, 34, 39, 44, 49, 54, 59, 64, 69, 74, 79, 84], 
    labels = ["18-24", "25-29", "30-34", "35-39", "40-44", "45-49", "50-54", "55-59", "60-64", "65-69", "70-74", "75-79", "80-84"],
)

C:\Users\brisa\AppData\Local\Temp\ipykernel_19316\1377674355.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  edad["edad_agrupada"] = pd.cut(


In [36]:
edad = edad.groupby(["edad_agrupada"]).count()

In [37]:
edad = edad.reset_index()

In [38]:
grafico_edad = alt.Chart(edad).mark_bar(color = "#8267BE").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("edad_agrupada:O",
         title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")],
).properties(
    title = "Edad de las informantes organizada por grupo",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(
    width=600, height=300,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [39]:
grafico_edad

alt.Chart(...)

Estado civil de las personas que se identificaron con sexo femenino 
1. Persona == **ID_PER**
2. Pregunta == Actualmente ¿usted... == **P1_7**
* **1** == está con su pareja en unión libre?
* **2** == está separada de una unión libre? 
* **3** == está separada de un matrimonio?
* **4** == está casada?
* **5** == está soltera?
* **6** == está divorciada?
* **7** == está viuda?
* **8** == No sabe 
* **9** == No responde

In [40]:
estado_civil = enpol_expectativas_mujeres[["ID_PER", "P1_7"]]

In [41]:
estado_civil_no_sabe = estado_civil["P1_7"] == 8

In [42]:
estado_civil_no_responde = estado_civil["P1_7"] == 9

In [43]:
estado_civil[~ estado_civil_no_sabe & ~ estado_civil_no_responde]["P1_7"].describe()

count    11689.000000
mean         3.387972
std          1.868485
min          1.000000
25%          1.000000
50%          4.000000
75%          5.000000
max          7.000000
Name: P1_7, dtype: float64

In [44]:
estado_civil = estado_civil[~ estado_civil_no_sabe & ~ estado_civil_no_responde].groupby(["P1_7"]).count()

In [45]:
estado_c = ["A. con pareja en union libre", "B. separada de una union libre", "C. separada de un matrimonio", "D. casada", "E. soltera", "E. divorciada", "F. viuda"]

In [46]:
estado_civil["estado_civil_mujeres"] = estado_c

In [47]:
primera_columna = estado_civil.pop("estado_civil_mujeres")
estado_civil.insert(0, "estado_civil_mujeres", primera_columna)
estado_civil

estado_civil_mujeres  ID_PER
P1_7                                        
1       A. con pareja en union libre    3189
2     B. separada de una union libre    1374
3       C. separada de un matrimonio     789
4                          D. casada    1923
5                         E. soltera    3466
6                      E. divorciada     360
7                           F. viuda     588

In [48]:
grafico_estado_civil = alt.Chart(estado_civil).mark_bar(color = "#B80D57").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("estado_civil_mujeres:N",
          title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Numero de informantes")],
).properties(
    title = "Estado civil de las informantes",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=600, height = 190,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [49]:
grafico_estado_civil

alt.Chart(...)

Madres con hijos menores de 18 años 

1. Persona == **ID_PER**
2. Pregunta == ¿Usted tiene hijos(as) menores de 18 años? == **P1_8**
* **1** == Sí
* **2** == No
* **8** == No sabe 
* **9** == No responde 
3. Pregunta == ¿Cuántos hijos(as) menores de 18 años tiene? == **P1_9**
* **01...97** == Número de Hijos 
* **98** == No sabe
* **99** == No responde
* **NaN** == No aplica 


In [50]:
madres_con_menores = enpol_expectativas_mujeres[["ID_PER", "P1_8", "P1_9"]].dropna()

In [51]:
mujeres_sin_hijos = (madres_con_menores["P1_8"] == 2) | (madres_con_menores["P1_8"] == 8) | (madres_con_menores["P1_8"] == 9)

In [52]:
numero_de_hijos_nulo = (madres_con_menores["P1_9"] == 98) | (madres_con_menores["P1_9"] == 99)

In [53]:
madres_con_menores[~ mujeres_sin_hijos & ~ numero_de_hijos_nulo]["P1_9"].describe()

count    7944.000000
mean        2.038646
std         1.073829
min         1.000000
25%         1.000000
50%         2.000000
75%         3.000000
max         9.000000
Name: P1_9, dtype: float64

In [54]:
madres_con_menores = madres_con_menores[~ mujeres_sin_hijos & ~ numero_de_hijos_nulo].sample(5000)

In [55]:
madres_con_menores

ID_PER  P1_8  P1_9
2484    777.5569     1   1.0
1669    549.7090     1   3.0
1565    519.6509     1   1.0
4650   1341.9139     1   4.0
5642   1602.7441     1   2.0
...          ...   ...   ...
2852    871.7371     1   1.0
9350   2579.6531     1   1.0
1649    544.5191     1   2.0
10320  2830.1973     1   2.0
9123   2515.3121     1   3.0

[5000 rows x 3 columns]

In [56]:
grafico_madres_con_menores = alt.Chart(madres_con_menores).mark_point(filled = True, size = 85, color = "#B80D57" ).encode(
    alt.X("P1_9:Q", title = "Numero de hijos o hojas", axis = alt.Axis(format = "1s", labelFontSize = 13)), 
    alt.Y("count()", title = "Madres entrevistadas", axis = alt.Axis(format = "1s", labelFontSize = 13)),
    tooltip = [alt.Tooltip("count()", title = "Madres entrevistadas")]
).properties(
    title = "Número de hijos o hijas menores de edad por madres entrevistadas",
    width = 400,
).configure_title( 
  offset = 15, fontSize = 14,
)

In [57]:
grafico_madres_con_menores

alt.Chart(...)

Cuidado de los hijos e hijas de mujeres en prisión

1. Persona == **ID_PER**
2. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? Yo (informante) == **P1_11_1**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
3. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? El padre == **P1_11_2**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
4. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? Abuelo(a)s == **P1_11_3**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
5.  Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? Otros familiares == **P1_11_4**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
6. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? Albergue, casa hogar o cuna u otra institución pública o privada de asistencia social == **P1_11_5**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
7. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? Nadie == **P1_11_6**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
8. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? Otro == **P1_11_7**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
9. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? No sabe == **P1_11_8**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica
10. Pregunta == Actualmente, ¿quién(es) se hace(n)  responsable(s) del cuidado de su(s) hijo(s) menor(es) de 18 años? No responde == **P1_11_9**
* **0** == No se declaró como respuesta afirmativa
* **1** == Sí
* **NaN** == No aplica

In [58]:
cuidado_hijos_madre = enpol_expectativas_mujeres[["ID_PER", "P1_11_1"]]
mask_cuidado_hijos_madre = cuidado_hijos_madre["P1_11_1"] == 1
cuidado_hijos_madre_f = cuidado_hijos_madre[mask_cuidado_hijos_madre]
cuidado_hijos_madre_f = cuidado_hijos_madre_f.groupby("P1_11_1").count()
cuidado_hijos_madre_f

ID_PER
P1_11_1        
1.0         215

In [59]:
cuidado_hijos_padre = enpol_expectativas_mujeres[["ID_PER", "P1_11_2"]]
mask_cuidado_hijos_padre = cuidado_hijos_padre["P1_11_2"] == 1
cuidado_hijos_padre_f = cuidado_hijos_padre[mask_cuidado_hijos_padre]
cuidado_hijos_padre_f = cuidado_hijos_padre_f.groupby("P1_11_2").count()
cuidado_hijos_padre_f

ID_PER
P1_11_2        
1.0        2444

In [60]:
cuidado_hijos_abuelos = enpol_expectativas_mujeres[["ID_PER", "P1_11_3"]]
mask_cuidado_hijos_abuelos = cuidado_hijos_abuelos["P1_11_3"] == 1
cuidado_hijos_abuelos_f = cuidado_hijos_abuelos[mask_cuidado_hijos_abuelos]
cuidado_hijos_abuelos_f = cuidado_hijos_abuelos_f.groupby("P1_11_3").count()
cuidado_hijos_abuelos_f

ID_PER
P1_11_3        
1.0        4387

In [61]:
cuidado_hijos_otros_fam = enpol_expectativas_mujeres[["ID_PER", "P1_11_4"]]
mask_cuidado_hijos_otros_fam = cuidado_hijos_otros_fam["P1_11_4"] == 1
cuidado_hijos_otros_fam_f = cuidado_hijos_otros_fam[mask_cuidado_hijos_otros_fam]
cuidado_hijos_otros_fam_f = cuidado_hijos_otros_fam_f.groupby("P1_11_4").count()
cuidado_hijos_otros_fam_f

ID_PER
P1_11_4        
1.0        1935

In [62]:
cuidado_hijos_institucion = enpol_expectativas_mujeres[["ID_PER", "P1_11_5"]]
mask_cuidado_hijos_institucion = cuidado_hijos_institucion["P1_11_5"] == 1
cuidado_hijos_institucion_f = cuidado_hijos_institucion[mask_cuidado_hijos_institucion]
cuidado_hijos_institucion_f = cuidado_hijos_institucion_f.groupby("P1_11_5").count()
cuidado_hijos_institucion_f

ID_PER
P1_11_5        
1.0         183

In [63]:
cuidado_hijos_nadie = enpol_expectativas_mujeres[["ID_PER", "P1_11_6"]]
mask_cuidado_hijos_nadie = cuidado_hijos_nadie["P1_11_6"] == 1
cuidado_hijos_nadie_f = cuidado_hijos_nadie[mask_cuidado_hijos_nadie]
cuidado_hijos_nadie_f = cuidado_hijos_nadie_f.groupby("P1_11_6").count()
cuidado_hijos_nadie_f

ID_PER
P1_11_6        
1.0          68

In [64]:
cuidado_hijos_otro = enpol_expectativas_mujeres[["ID_PER", "P1_11_7"]]
mask_cuidado_hijos_otro = cuidado_hijos_otro["P1_11_7"] == 1
cuidado_hijos_otro_f = cuidado_hijos_otro[mask_cuidado_hijos_otro]
cuidado_hijos_otro_f = cuidado_hijos_otro_f.groupby("P1_11_7").count()
cuidado_hijos_otro_f

ID_PER
P1_11_7        
1.0          80

In [65]:
cuidado_hijos_no_sabe = enpol_expectativas_mujeres[["ID_PER", "P1_11_8"]]
mask_cuidado_hijos_no_sabe = cuidado_hijos_no_sabe["P1_11_8"] == 1
cuidado_hijos_no_sabe_f = cuidado_hijos_no_sabe[mask_cuidado_hijos_no_sabe]
cuidado_hijos_no_sabe_f = cuidado_hijos_no_sabe_f.groupby("P1_11_8").count()
cuidado_hijos_no_sabe_f

ID_PER
P1_11_8        
1.0          21

In [66]:
cuidado_de_hijos = pd.concat([cuidado_hijos_madre_f, cuidado_hijos_padre_f, cuidado_hijos_abuelos_f, cuidado_hijos_otros_fam_f, cuidado_hijos_institucion_f, cuidado_hijos_nadie_f, cuidado_hijos_otro_f,  cuidado_hijos_no_sabe_f], axis=0)

In [67]:
encargados = ["A. madre", "B. padre", "C. abuelos", "D. otros familiares", "E. institucion de asistencia social", "F. nadie", "G. otro", "H. No sabe"]

In [68]:
cuidado_de_hijos["Cuidado_de_menores"] = encargados

In [69]:
primera_columna = cuidado_de_hijos.pop("Cuidado_de_menores")
cuidado_de_hijos.insert(0, "Cuidado_de_menores", primera_columna)
cuidado_de_hijos

Cuidado_de_menores  ID_PER
1.0                             A. madre     215
1.0                             B. padre    2444
1.0                           C. abuelos    4387
1.0                  D. otros familiares    1935
1.0  E. institucion de asistencia social     183
1.0                             F. nadie      68
1.0                              G. otro      80
1.0                           H. No sabe      21

In [70]:
grafico_cuidado_de_hijos = alt.Chart(cuidado_de_hijos).mark_bar(color = "#B80D57").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("Cuidado_de_menores:N",
          title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "respuestas")],
).properties(
    title = "Personas al cuidado de los hijos menores de edad de las madres entrevistadas",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=600, height=250,
).configure_title( 
  offset = 15, fontSize = 16,
)

In [71]:
grafico_cuidado_de_hijos

alt.Chart(...)

Pertenencia étnica
1. Persona == **ID_PER**
2. Pregunta == De acuerdo con su cultura, historia y tradiciones... == **P1_15**
* **1** == ¿se considera negro(a), es decir, afromexicano(a) o afrodescendiente?
* **2** == ¿se considera indígena?
* **3** == Ninguna de las anteriores 
* **8** == No sabe
* **9** == No responde

In [72]:
pertenencia_etnica = enpol_expectativas_mujeres[["ID_PER", "P1_15"]]

In [73]:
pertenencia_etnica_no_sabe = pertenencia_etnica["P1_15"] == 8

In [74]:
pertenencia_etnica_no_responde = pertenencia_etnica["P1_15"] == 9

In [75]:
pertenencia_etnica[~ pertenencia_etnica_no_sabe & ~ pertenencia_etnica_no_responde]["P1_15"].describe()

count    11570.000000
mean         2.829386
std          0.414243
min          1.000000
25%          3.000000
50%          3.000000
75%          3.000000
max          3.000000
Name: P1_15, dtype: float64

In [76]:
pertenencia_etnica_filtrado = pertenencia_etnica[~ pertenencia_etnica_no_sabe & ~ pertenencia_etnica_no_responde].groupby(["P1_15"]).count()

In [77]:
pertenecia = ["Afrodescendiente", "Indígena", "Ninguna mencionada"]

In [78]:
pertenencia_etnica_filtrado["pertenecia_e"] = pertenecia

In [79]:
primera_columna = pertenencia_etnica_filtrado.pop("pertenecia_e")
pertenencia_etnica_filtrado.insert(0, "pertenencia_e", primera_columna)
pertenencia_etnica_filtrado

pertenencia_e  ID_PER
P1_15                            
1        Afrodescendiente     174
2                Indígena    1626
3      Ninguna mencionada    9770

In [80]:
grafico_pertenencia_etnica = alt.Chart(pertenencia_etnica_filtrado).mark_bar(color = "#DBBF0D").encode(
    alt.X("ID_PER:Q", axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("pertenencia_e:N", axis = alt.Axis(grid = False, labelFontSize = 13),
         title = " "),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Informantes")],
).properties(
    title = "Pertenencia etnica de acuerdo con cultura, historia y tradiciones",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=600, height=100,
).configure_title( 
  offset = 15, fontSize = 16,
)

In [81]:
grafico_pertenencia_etnica

alt.Chart(...)

Escolaridad 
1. Persona == **ID_PER**
2. Pregunta == ¿Hasta qué año o grado aprobó la escuela? Nivel == **P1_18_N**
* **0** == Ninguno 
* **1** == Preescolar 
* **2** == Primaria 
* **3** == Secundaria
* **4** == Carrera técnica con secundaria terminada 
* **5** == Normal básica (con antecedente en secundaria)
* **6** == Preparatoria o bachillerato
* **7** == Carrera técnica con preparatoria terminada 
* **8** == Licenciatura o profesional 
* **9** == Maestría o doctorado 
* **98** == No sabe
* **99** == No responde 


In [82]:
escolaridad = enpol_expectativas_mujeres[["ID_PER", "P1_18_N"]]

In [83]:
escolaridad_no_sabe = escolaridad["P1_18_N"] == 98

In [84]:
escolaridad_no_responde = escolaridad["P1_18_N"] == 99

In [85]:
escolaridad[~ escolaridad_no_sabe & ~ escolaridad_no_responde]["P1_18_N"].describe()

count    11699.000000
mean         4.034105
std          2.045586
min          0.000000
25%          3.000000
50%          3.000000
75%          6.000000
max          9.000000
Name: P1_18_N, dtype: float64

In [86]:
escolaridad = escolaridad[~ escolaridad_no_sabe & ~ escolaridad_no_responde].groupby(["P1_18_N"]).count()

In [87]:
escolaridad_nivel = ["A. ninguno", "B. preescolar", "C. primaria", "D. secundaria", "E. carrera técnica con secundaria", "F. normal básica antec. secundaria", "G. preparatoria", "H. carrera técnica con preparatoria", "I. licenciatura o profesional", "J. maestria o doctorado"]

In [88]:
escolaridad["nivel_escolar"] = escolaridad_nivel

In [89]:
primera_columna = escolaridad.pop("nivel_escolar")
escolaridad.insert(0, "nivel_escolar", primera_columna)
escolaridad

nivel_escolar  ID_PER
P1_18_N                                             
0                                 A. ninguno     290
1                              B. preescolar      12
2                                C. primaria    1872
3                              D. secundaria    5283
4          E. carrera técnica con secundaria     204
5         F. normal básica antec. secundaria      18
6                            G. preparatoria    2644
7        H. carrera técnica con preparatoria     263
8              I. licenciatura o profesional    1038
9                    J. maestria o doctorado      75

In [90]:
grafico_escolaridad = alt.Chart(escolaridad).mark_bar(color = "#DBBF0D").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("nivel_escolar:O",
         title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Informantes")],
).properties(
    title = "Escolaridad de acuerdo al último nivel de estudios concluido",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=600, height=300,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [91]:
grafico_escolaridad

alt.Chart(...)

Identidad 
1. Persona == **ID_PER**
2. Pregunta == ¿Cómo se identifica usted? == **P1_22**
* **1** == Como hombre 
* **2** == Como mujer
* **3** == Como mujer trans (transexual, transgénero o transvesti)
* **4** == Como hombre trans (transexual, transgénero o transvesti)
* **5** == Otra 
* **6** == Prefiero no responder
* **8** == No entendí

In [92]:
identidad = enpol_expectativas_mujeres[["ID_PER", "P1_22"]]

In [93]:
identidad_no_responde = identidad["P1_22"] == 6

In [94]:
identidad_no_entendio = identidad["P1_22"] == 8

In [95]:
identidad[~ identidad_no_responde & ~ identidad_no_entendio].describe()

ID_PER         P1_22
count  11693.000000  11693.000000
mean    1654.461607      2.006243
std      891.900212      0.197605
min      101.143200      1.000000
25%      891.327400      2.000000
50%     1646.228200      2.000000
75%     2425.800300      2.000000
max     3203.333600      4.000000

In [96]:
identidad = identidad[~ identidad_no_responde & ~ identidad_no_entendio].groupby("P1_22").count()

In [97]:
identidad_respuesta = ["Hombre", "Mujer", "Mujer trans", "Hombre trans"]

In [98]:
identidad["respuesta"] = identidad_respuesta

In [99]:
primera_columna = identidad.pop("respuesta")
identidad.insert(0, "respuesta", primera_columna)
identidad

respuesta  ID_PER
P1_22                      
1            Hombre     177
2             Mujer   11281
3       Mujer trans     220
4      Hombre trans      15

In [100]:
grafico_identidad = alt.Chart(identidad).mark_bar(color = "#FFD868").encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Informantes")]
).properties(
    title = "Identidad",
    width = 300, height = 300,
)

In [101]:
grafico_identidad

alt.Chart(...)

Orientación sexual 
1. Persona == **ID_PER**
2. Pregunta == De las siguientes frases, ¿cuál define su orientación sexual? == **P1_23**
* **1** == Me gustan las personas de ambos sexos (Bisexual)
* **2** == Me gustan las personas de mi mismo sexo (Homosexual)
* **3** == Me gustan las personas del sexo opuesto (Heterosexual)
* **4** == Otra 
* **8** == No sabe 
* **9** == No responde

In [102]:
orientacion = enpol_expectativas_mujeres[["ID_PER", "P1_23"]]

In [103]:
orientacion_no_sabe = orientacion["P1_23"] == 8

In [104]:
orientacion_no_responde = orientacion["P1_23"] == 9

In [105]:
orientacion[~ orientacion_no_sabe & ~ orientacion_no_responde].describe()

ID_PER         P1_23
count  11676.000000  11676.000000
mean    1655.000969      2.686451
std      892.110182      0.688797
min      101.143200      1.000000
25%      891.559825      3.000000
50%     1646.771350      3.000000
75%     2427.361600      3.000000
max     3203.333600      4.000000

In [106]:
orientacion = orientacion[~ orientacion_no_sabe & ~ orientacion_no_responde].groupby("P1_23").count()

In [107]:
orientacion_respuesta = ["Bisexual", "Homosexual", "Heterosexual", "Otra"]

In [108]:
orientacion["respuesta"] = orientacion_respuesta

In [109]:
primera_columna = orientacion.pop("respuesta")
orientacion.insert(0, "respuesta", primera_columna)
orientacion

respuesta  ID_PER
P1_23                      
1          Bisexual    1505
2        Homosexual     659
3      Heterosexual    9504
4              Otra       8

In [110]:
grafico_orientacion = alt.Chart(orientacion).mark_bar(color = "#FFD868").encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Informantes")]
).properties(
    title = "Orientación sexual",
    width = 300, height = 300,
)

In [111]:
grafico_orientacion

alt.Chart(...)

In [112]:
grafico_orientacion_identidad = grafico_identidad | grafico_orientacion

In [113]:
grafico_orientacion_identidad

alt.HConcatChart(...)

## Proceso judicial 

Tiempo que permaneceran en prisión las personas sentenciadas 
1. Persona == **ID_PER**
2. Pregunta == ¿Cuánto tiempo le dijo el juez que permanecería recluido en un Centro Penitenciario? Años == **P5_4_A**
* **01...96** == Años
* **00** == Casillas vacías
* **97** == 97 años o más
* **98** == No sabe 
* **99** == No responde
* **NaN** == No aplica

In [114]:
tiempo_en_prision_sentenciadas = enpol_expectativas_mujeres[["ID_PER", "P5_4_A"]].dropna()

In [115]:
anios_prision_noventaysiete_mas = tiempo_en_prision_sentenciadas["P5_4_A"] == 97

In [116]:
anios_prision_no_sabe = tiempo_en_prision_sentenciadas["P5_4_A"] == 98

In [117]:
anios_prision_no_responde = tiempo_en_prision_sentenciadas["P5_4_A"] == 99

In [118]:
tiempo_en_prision_sentenciadas[~ anios_prision_no_sabe & ~ anios_prision_no_responde]["P5_4_A"].describe()

count    6229.000000
mean       24.136459
std        21.102960
min         0.000000
25%         7.000000
50%        19.000000
75%        35.000000
max        97.000000
Name: P5_4_A, dtype: float64

In [119]:
tiempo_en_prision_sentenciadas = tiempo_en_prision_sentenciadas[~ anios_prision_no_sabe & ~ anios_prision_no_responde].sample(5000)

In [120]:
grafico_anios_prision_sentenciadas = alt.Chart(tiempo_en_prision_sentenciadas).mark_point(filled = True, size = 50, color = "#8267BE" ).encode(
    alt.X("P5_4_A:Q", title = "Años de sentencia"), 
    alt.Y("count()", axis = alt.Axis(format = "1s"), title = "Número de mujeres sentenciadas"),
    tooltip = [alt.Tooltip("count()", title = "Número de mujeres sentenciadas"), alt.Tooltip("P5_4_A", title = "Anios")],
).properties(
    title = "Años que permaneceran en prision las mujeres a las cuales se les ha dictado sentencia",
    width = 800,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [121]:
grafico_anios_prision_sentenciadas

alt.Chart(...)

Tiempo que transcurrio para que a una persona le dictaran setencia 
1. == Persona == **ID_PER**
2. Pregunta == Desde su llegada al centro penitenciario ¿cuánto tiempo transcurrió para que el Juez el dictara sentencia? == **P5_10**
* **1** == Hasta una semana 
* **2** == Más de una semana hasta dos semanas 
* **3** == Más de dos semanas hasta un mes
* **4** == Más de un mes hasta seis meses
* **5** == Más de seis meses hasta un año 
* **6** == Más de un año hasta dos años 
* **7** == Más de dos años
* **8** == No sabe 
* **9** == No responde 
* **NaN** == No aplica

In [122]:
tiempo_transcurrido_sentencia = enpol_expectativas_mujeres[["ID_PER", "P5_10"]].dropna()

In [123]:
tiempo_transcurrido_sentencia_no_sabe = tiempo_transcurrido_sentencia["P5_10"] == 8

In [124]:
tiempo_transcurrido_sentencia_no_responde = tiempo_transcurrido_sentencia["P5_10"] == 9

In [125]:
tiempo_transcurrido_sentencia[~ tiempo_transcurrido_sentencia_no_sabe & ~ tiempo_transcurrido_sentencia_no_responde]["P5_10"].describe()

count    6062.000000
mean        5.517816
std         1.290989
min         1.000000
25%         5.000000
50%         6.000000
75%         7.000000
max         7.000000
Name: P5_10, dtype: float64

In [126]:
tiempo_transcurrido_sentencia = tiempo_transcurrido_sentencia[~ tiempo_transcurrido_sentencia_no_sabe & ~ tiempo_transcurrido_sentencia_no_responde].groupby(["P5_10"]).count()

In [127]:
tiempo = ["A. hasta una semana", "B. mas de una semana hasta dos", "C. mas de dos semanas hasta un mes", "D. mas de un mes hasta seis meses", "E. mas de seis meses hasta un año", "F. mas de un año hasta dos años", "G. mas de dos años"]

In [128]:
tiempo_transcurrido_sentencia["Tiempo que transcurrio para que el juez dictara sentencia"] = tiempo

In [129]:
primera_columna = tiempo_transcurrido_sentencia.pop("Tiempo que transcurrio para que el juez dictara sentencia")
tiempo_transcurrido_sentencia.insert(0, "Tiempo que transcurrio para que el juez dictara sentencia", primera_columna)
tiempo_transcurrido_sentencia

Tiempo que transcurrio para que el juez dictara sentencia  ID_PER
P5_10                                                                  
1.0                                  A. hasta una semana            113
2.0                       B. mas de una semana hasta dos             30
3.0                   C. mas de dos semanas hasta un mes             98
4.0                    D. mas de un mes hasta seis meses           1016
5.0                    E. mas de seis meses hasta un año           1586
6.0                      F. mas de un año hasta dos años           1545
7.0                                   G. mas de dos años           1674

In [130]:
grafico_tiempo_transcurrido_sentencia = alt.Chart(tiempo_transcurrido_sentencia).mark_bar(color = "#DBBF0D").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("Tiempo que transcurrio para que el juez dictara sentencia:O",
          title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Numero de informantes")],
).properties(
    title = "Tiempo que transcurrio para le fuera dictada sentencia",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=700, height=200,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [131]:
grafico_tiempo_transcurrido_sentencia

alt.Chart(...)

Tiempo que ha durado el proceso o juicio de las personas NO sentenciadas 
1. Persona == **ID_PER**
2. Aproximadamente ¿cuánto ha durado su proceso o juicio? Años == **P5_34_A**
* **01...95** == Años
* **00** == Casillas vacías 
* **96** == 29 días o menos
* **98** == No sabe 
* **99** == No responde
* **NaN** == No aplica 

In [132]:
tiempo_proceso_no_sentenciadas = enpol_expectativas_mujeres[["ID_PER","P5_34_A"]].dropna()

In [133]:
dias_proceso_no_sentenciadas = tiempo_proceso_no_sentenciadas["P5_34_A"] == 96

In [134]:
anios_proceso_no_sabe = tiempo_proceso_no_sentenciadas["P5_34_A"] == 98

In [135]:
anios_proceso_no_responde = tiempo_proceso_no_sentenciadas["P5_34_A"] == 99

In [136]:
tiempo_proceso_no_sentenciadas[~ anios_proceso_no_sabe & ~ anios_proceso_no_responde & ~ dias_proceso_no_sentenciadas]["P5_34_A"].describe()

count    5066.000000
mean        2.106790
std         3.170671
min         0.000000
25%         0.000000
50%         1.000000
75%         3.000000
max        22.000000
Name: P5_34_A, dtype: float64

In [137]:
tiempo_proceso_no_sentenciadas_meses_años = tiempo_proceso_no_sentenciadas[~ anios_proceso_no_sabe & ~ anios_proceso_no_responde & ~ dias_proceso_no_sentenciadas].sample(5000)

In [138]:
grafico_tiempo_proceso_no_sentenciadas = alt.Chart(tiempo_proceso_no_sentenciadas_meses_años).mark_point(filled = True, size = 50, color = "#8267BE" ).encode(
    alt.X("P5_34_A:Q", title = "Años de proceso"), 
    alt.Y("count()", axis = alt.Axis(format = "1s"), title = "Número de mujeres en proceso"),
    tooltip = [alt.Tooltip("count()", title = "Número de mujeres en proceso"), alt.Tooltip("P5_34_A", title = "Anios")]
).properties(
    title = "Años que lleva el proceso de las mujeres que aún no son sentenciadas",
    width = 800,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [139]:
grafico_tiempo_proceso_no_sentenciadas

alt.Chart(...)

## Expectativas de salida 

Sobre la reducción de sentencia
1. Persona == **ID_PER**
2. Pregunta == ¿Usted ha obtenido algún beneficio para reducir su sentencia? == **P10_1**
* **1** == Sí 
* **2** == No
* **8** == No sabe 
* **9** == No reponde 
* **NaN** == No aplica 

In [140]:
reducir_sentencia = enpol_expectativas_mujeres[["ID_PER", "P10_1"]].dropna()

In [141]:
reducir_sentencia_no_sabe = reducir_sentencia["P10_1"] == 8

In [142]:
reducir_sentencia_no_responde = reducir_sentencia["P10_1"] == 9

In [143]:
reducir_sentencia = reducir_sentencia[~ reducir_sentencia_no_sabe & ~ reducir_sentencia_no_responde].groupby(["P10_1"]).count()

In [144]:
reduccion_sentencia = ["Si", "No"]

In [145]:
reducir_sentencia["respuesta"] = reduccion_sentencia

In [146]:
primera_columna = reducir_sentencia.pop("respuesta")
reducir_sentencia.insert(0, "respuesta", primera_columna)
reducir_sentencia

respuesta  ID_PER
P10_1                  
1.0          Si    1313
2.0          No    4894

In [147]:
grafico_reducir_sentencia = alt.Chart(reducir_sentencia).mark_bar().encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False, labelFontSize = 12)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False, labelFontSize = 12)),
    alt.Color("respuesta", scale = alt.Scale(domain = ["Si", "No"], range = ["#FFCD19", "#8267BE"])),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")]
).properties(
    title = "¿Usted ha obtenido algún beneficio para reducir su sentencia?",
    width = 150, height = 400,
).configure_title(
  anchor = "start", 
  offset = 15, fontSize = 14,
)

In [148]:
grafico_reducir_sentencia

alt.Chart(...)

Vivienda al salir del Centro Penitenciario
1. Persona == **ID_PER**
2. Pregunta == Pensando en el momento de su salida de este Centro, ¿tendría a dónde llegar a vivir, aunque sea temporalmente? == **P10_2**
* **1** == Sí 
* **2** == No
* **8** == No sabe 
* **9** == No reponde  

In [149]:
vivienda_al_salir = enpol_expectativas_mujeres[["ID_PER", "P10_2"]]

In [150]:
vivienda_al_salir_no_sabe = vivienda_al_salir["P10_2"] == 8

In [151]:
vivienda_al_salir_no_responde = vivienda_al_salir["P10_2"] == 9

In [152]:
vivienda_al_salir = vivienda_al_salir[~ vivienda_al_salir_no_sabe & ~ vivienda_al_salir_no_responde].groupby(["P10_2"]).count()

In [153]:
vivienda = ["Si", "No"]

In [154]:
vivienda_al_salir["respuesta"] = vivienda

In [155]:
primera_columna = vivienda_al_salir.pop("respuesta")
vivienda_al_salir.insert(0, "respuesta", primera_columna)
vivienda_al_salir

respuesta  ID_PER
P10_2                  
1            Si   11190
2            No     473

In [156]:
grafico_vivienda_al_salir = alt.Chart(vivienda_al_salir).mark_bar().encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False, labelFontSize = 12)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False, labelFontSize = 12)),
    alt.Color("respuesta", scale = alt.Scale(domain = ["Si", "No"], range = ["#FFCD19", "#8267BE"])),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")],
).properties(
    title = "Al momento de su salida de este Centro, ¿tendría a dónde llegar a vivir?",
    width = 150, height = 400,
).configure_title(
  anchor = "start", 
  offset = 15 
).configure_title(
  anchor = "start", 
  offset = 15, fontSize = 14,
)

In [157]:
grafico_vivienda_al_salir

alt.Chart(...)

Causas por las cuales algunas personas expresaron que no tendrían vivienda al momento de salir de prisión
1. Persona == **ID_PER**
2. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? Porque la rechazarían las personas con quienes vivía == **P10_4_1**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica 
3. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? Porque las personas con quienes vivía se han mudado == **P10_4_2**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica 
4. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? Porque no tendría para pagar un lugar == **P10_4_3**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica 
5. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? Porque ha perdido contacto con sus conocidos == **P10_4_4**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica
6. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? Porque no quiere afectar a su familia == **P10_4_5**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica
7. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? == Otra razón == **P10_4_6**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica
8. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? == No sabe == **P10_4_8**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica
9. Pregunta == ¿Por qué considera que no tendría a dónde llegar a vivir? == No responde == **P10_4_9**
* **0** == No se declaró como respuesta afirmativa 
* **1** == Sí 
* **NaN** No aplica

In [158]:
causa_no_vivienda_rechazo = enpol_expectativas_mujeres[["ID_PER", "P10_4_1"]]
mask_causa_no_vivienda_rechazo = causa_no_vivienda_rechazo["P10_4_1"] == 1
causa_no_vivienda_rechazo_f = causa_no_vivienda_rechazo[mask_causa_no_vivienda_rechazo]
causa_no_vivienda_rechazo_f = causa_no_vivienda_rechazo_f.groupby("P10_4_1").count()
causa_no_vivienda_rechazo_f

ID_PER
P10_4_1        
1.0          97

In [159]:
causa_no_vivienda_cambio_domicilio = enpol_expectativas_mujeres[["ID_PER", "P10_4_2"]]
mask_causa_no_vivienda_cambio_domicilio = causa_no_vivienda_cambio_domicilio["P10_4_2"] == 1
causa_no_vivienda_cambio_domicilio_f = causa_no_vivienda_cambio_domicilio[mask_causa_no_vivienda_cambio_domicilio]
causa_no_vivienda_cambio_domicilio_f = causa_no_vivienda_cambio_domicilio_f.groupby("P10_4_2").count()
causa_no_vivienda_cambio_domicilio_f

ID_PER
P10_4_2        
1.0          32

In [160]:
causa_no_vivienda_sin_dinero = enpol_expectativas_mujeres[["ID_PER", "P10_4_3"]]
mask_causa_no_vivienda_sin_dinero = causa_no_vivienda_sin_dinero["P10_4_3"] == 1
causa_no_vivienda_sin_dinero_f = causa_no_vivienda_sin_dinero[mask_causa_no_vivienda_sin_dinero]
causa_no_vivienda_sin_dinero_f = causa_no_vivienda_sin_dinero_f.groupby("P10_4_3").count()
causa_no_vivienda_sin_dinero_f

ID_PER
P10_4_3        
1.0         104

In [161]:
causa_no_vivienda_sin_contactos = enpol_expectativas_mujeres[["ID_PER", "P10_4_4"]]
mask_causa_no_vivienda_sin_contactos = causa_no_vivienda_sin_contactos["P10_4_4"] == 1
causa_no_vivienda_sin_contactos_f = causa_no_vivienda_sin_contactos[mask_causa_no_vivienda_sin_contactos]
causa_no_vivienda_sin_contactos_f = causa_no_vivienda_sin_contactos_f.groupby("P10_4_4").count()
causa_no_vivienda_sin_contactos_f

ID_PER
P10_4_4        
1.0          54

In [162]:
causa_no_vivienda_no_afectarfam = enpol_expectativas_mujeres[["ID_PER", "P10_4_5"]]
mask_causa_no_vivienda_no_afectarfam = causa_no_vivienda_no_afectarfam["P10_4_5"] == 1
causa_no_vivienda_no_afectarfam_f = causa_no_vivienda_no_afectarfam[mask_causa_no_vivienda_no_afectarfam]
causa_no_vivienda_no_afectarfam_f = causa_no_vivienda_no_afectarfam_f.groupby("P10_4_5").count()
causa_no_vivienda_no_afectarfam_f

ID_PER
P10_4_5        
1.0         108

In [163]:
causa_no_vivienda_otra_razon = enpol_expectativas_mujeres[["ID_PER", "P10_4_6"]]
mask_causa_no_vivienda_otra_razon = causa_no_vivienda_otra_razon["P10_4_6"] == 1
causa_no_vivienda_otra_razon_f = causa_no_vivienda_otra_razon[mask_causa_no_vivienda_otra_razon]
causa_no_vivienda_otra_razon_f = causa_no_vivienda_otra_razon_f.groupby("P10_4_6").count()
causa_no_vivienda_otra_razon_f

ID_PER
P10_4_6        
1.0         114

In [164]:
causas_no_vivienda = pd.concat([causa_no_vivienda_rechazo_f, causa_no_vivienda_cambio_domicilio_f, causa_no_vivienda_sin_dinero_f, causa_no_vivienda_sin_contactos_f, causa_no_vivienda_no_afectarfam_f, causa_no_vivienda_otra_razon_f], axis=0)

In [165]:
causas = ["La rechazarían", "Se han mudado sus familiares", "No tendría para pagar un lugar", "Perdió contacto conconocidos", "No quiere afectar a su familia", "Otra razon"]

In [166]:
causas_no_vivienda["Causas por las que no tendria un lugar para llegar a vivir"] = causas

In [167]:
primera_columna = causas_no_vivienda.pop("Causas por las que no tendria un lugar para llegar a vivir")
causas_no_vivienda.insert(0, "Causas por las que no tendria un lugar para llegar a vivir", primera_columna)
causas_no_vivienda

Causas por las que no tendria un lugar para llegar a vivir  ID_PER
1.0                                     La rechazarían              97
1.0                       Se han mudado sus familiares              32
1.0                     No tendría para pagar un lugar             104
1.0                       Perdió contacto conconocidos              54
1.0                     No quiere afectar a su familia             108
1.0                                         Otra razon             114

In [168]:
grafico_causas_no_vivienda = alt.Chart(causas_no_vivienda).mark_bar(color = "#3FA796").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(grid=False, labelFontSize = 13),
         title = " "),
    alt.Y("Causas por las que no tendria un lugar para llegar a vivir:N",
         title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")],
).properties(
    title = "¿Por qué considera que no tendría a dónde llegar a vivir al salir de prisión?",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=600, height=200,
).configure_title( 
  offset = 15, fontSize = 15,
)

In [169]:
grafico_causas_no_vivienda

alt.Chart(...)

Posibles repercusiones en diferentes aspectos de la vida cotidiana al salir del Centro Penitenciario 
1. Persona == **ID_PER**
2. Pregunta == Pensando en el momento en el que usted esté en libertad ¿considera que el haber estado en un Centro penitenciario afecte sus posibilidades de encontrar trabajo? == **P10_5_1** 
* **1** == Sí
* **2** == No 
* **8** == No sabe 
* **9** == No responde 
3. Pregunta == Pensando en el momento en el que usted esté en libertad ¿considera que el haber estado en un Centro penitenciario afecte sus posibilidades de continuar estudiando? == **P10_5_2**
* **1** == Sí
* **2** == No
* **3** == No aplica 
* **8** == No sabe 
* **9** == No responde 
4. Pregunta == Pensando en el momento en el que usted esté en libertad ¿considera que el haber estado en un Centro penitenciario afecte sus posibilidades de reencontrar amigos? == **P10_5_3**
* **1** == Sí
* **2** == No
* **3** == No aplica 
* **8** == No sabe 
* **9** == No responde
5. Pregunta == Pensando en el momento en el que usted esté en libertad ¿considera que el haber estado en un Centro penitenciario afecte sus posibilidades de reintegrarse a su familia? == **P10_5_4**
* **1** == Sí
* **2** == No
* **3** == No aplica 
* **8** == No sabe 
* **9** == No responde

In [170]:
repercusiones_en_libertad_trabajo = enpol_expectativas_mujeres[["ID_PER", "P10_5_1"]]
mask_repercusiones_en_libertad_trabajo = repercusiones_en_libertad_trabajo["P10_5_1"] == 1
repercusiones_en_libertad_trabajo_f = repercusiones_en_libertad_trabajo[mask_repercusiones_en_libertad_trabajo]
repercusiones_en_libertad_trabajo_f = repercusiones_en_libertad_trabajo_f.groupby("P10_5_1").count()
repercusiones_en_libertad_trabajo_f

ID_PER
P10_5_1        
1          7125

In [171]:
repercusiones_en_libertad_estudios = enpol_expectativas_mujeres[["ID_PER", "P10_5_2"]]
mask_repercusiones_en_libertad_estudios = repercusiones_en_libertad_estudios["P10_5_2"] == 1
repercusiones_en_libertad_estudios_f = repercusiones_en_libertad_estudios[mask_repercusiones_en_libertad_estudios]
repercusiones_en_libertad_estudios_f = repercusiones_en_libertad_estudios_f.groupby("P10_5_2").count()
repercusiones_en_libertad_estudios_f

ID_PER
P10_5_2        
1          2827

In [172]:
repercusiones_en_libertad_amigos = enpol_expectativas_mujeres[["ID_PER", "P10_5_3"]]
mask_repercusiones_en_libertad_amigos = repercusiones_en_libertad_amigos["P10_5_3"] == 1
repercusiones_en_libertad_amigos_f = repercusiones_en_libertad_amigos[mask_repercusiones_en_libertad_amigos]
repercusiones_en_libertad_amigos_f = repercusiones_en_libertad_amigos_f.groupby("P10_5_3").count()
repercusiones_en_libertad_amigos_f

ID_PER
P10_5_3        
1          3643

In [173]:
repercusiones_en_libertad_familia = enpol_expectativas_mujeres[["ID_PER", "P10_5_4"]]
mask_repercusiones_en_libertad_familia = repercusiones_en_libertad_familia["P10_5_4"] == 1
repercusiones_en_libertad_familia_f = repercusiones_en_libertad_familia[mask_repercusiones_en_libertad_familia]
repercusiones_en_libertad_familia_f = repercusiones_en_libertad_familia_f.groupby("P10_5_4").count()
repercusiones_en_libertad_familia_f

ID_PER
P10_5_4        
1          3204

In [174]:
posibles_repercusiones_en_libertad = pd.concat([repercusiones_en_libertad_trabajo_f, repercusiones_en_libertad_estudios_f, repercusiones_en_libertad_amigos_f, repercusiones_en_libertad_familia_f], axis=0)

In [175]:
posibles_repercusiones = ["Encontrar trabajo", "Seguir estudiando", "Reencontrar a sus amigos", "Reintegrarse a su familia"]

In [176]:
posibles_repercusiones_en_libertad["Considera que haber estado en prision afecte sus posibilidades de"] = posibles_repercusiones

In [177]:
primera_columna = posibles_repercusiones_en_libertad.pop("Considera que haber estado en prision afecte sus posibilidades de")
posibles_repercusiones_en_libertad.insert(0, "Considera que haber estado en prision afecte sus posibilidades de", primera_columna)
posibles_repercusiones_en_libertad

Considera que haber estado en prision afecte sus posibilidades de  ID_PER
1                                  Encontrar trabajo                   7125
1                                  Seguir estudiando                   2827
1                           Reencontrar a sus amigos                   3643
1                          Reintegrarse a su familia                   3204

In [178]:
grafico_posibles_repercusiones = alt.Chart(posibles_repercusiones_en_libertad).mark_bar(color = "#3FA796").encode(
    alt.X("ID_PER:Q",
         axis = alt.Axis(format = "1s", grid = False, labelFontSize = 13),
         title = " "),
    alt.Y("Considera que haber estado en prision afecte sus posibilidades de:N",
         title = " ", axis = alt.Axis(labelFontSize = 13)),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")],
).properties(
    title = "Considera que haber estado en prision afecte sus posibilidades de:",
    width = 500,
).configure_view(
    strokeWidth = 0
).properties(width=700, height=150
).configure_title( 
  offset = 15, fontSize = 15,
)

In [179]:
grafico_posibles_repercusiones

alt.Chart(...)

Herramientas para la vida adquiridas en el Centro penitenciario para reincertarse a la vida en libertad
1. Persona == **ID_PER**
2. Pregunta == ¿Usted considera que el Centro penitenciario le ha dado las herramientas necesarias para reincorporarse a la vida social del exterior? == **P10_6**
* **1** == Sí
* **2** == No
* **8** == No sabe 
* **9** == No responde

In [180]:
herramientas_para_reinsercion = enpol_expectativas_mujeres[["ID_PER", "P10_6"]]

In [181]:
herramientas_para_reinsercion_no_sabe = herramientas_para_reinsercion["P10_6"] == 8

In [182]:
herramientas_para_reinsercion_no_responde = herramientas_para_reinsercion["P10_6"] == 9

In [183]:
herramientas_para_reinsercion = herramientas_para_reinsercion[~ herramientas_para_reinsercion_no_sabe & ~ herramientas_para_reinsercion_no_responde].groupby(["P10_6"]).count()

In [184]:
herramientas_reinsercion = ["Si", "No"]

In [185]:
herramientas_para_reinsercion["respuesta"] = herramientas_reinsercion

In [186]:
primera_columna = herramientas_para_reinsercion.pop("respuesta")
herramientas_para_reinsercion.insert(0, "respuesta", primera_columna)
herramientas_para_reinsercion

respuesta  ID_PER
P10_6                  
1            Si    7391
2            No    4174

In [187]:
grafico_herramientas_reinsercion = alt.Chart(herramientas_para_reinsercion).mark_bar().encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False, labelFontSize = 12)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False, labelFontSize = 12)),
    alt.Color("respuesta", scale = alt.Scale(domain = ["Si", "No"], range = ["#FFCD19", "#8267BE"])),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")]
).properties(
    title = "¿El Centro le ha dado las herramientas para reincorporarse a la vida del exterior?",
    width = 150, height = 400,
).configure_title(
  anchor = "start", 
  offset = 14 
)

In [188]:
grafico_herramientas_reinsercion

alt.Chart(...)

Sobre posible regreso a prisión 
1. Persona == **ID_PER**
2. Pregunta == Considerando los retos y obstáculos que usted puede enfrentar una vez saliendo del Centro penitenciario, ¿qué tan probable o improbable considera regresar a prisión? == **P10_7**
* **1** Muy probable 
* **2** Algo probable 
* **3** Algo improbable
* **4** Muy improbable 
* **8** No sabe
* **9** No responde 

In [189]:
posibilidad_regreso_prision = enpol_expectativas_mujeres[["ID_PER", "P10_7"]]

In [190]:
posibilidad_regreso_prision_no_sabe = posibilidad_regreso_prision["P10_7"] == 8

In [191]:
posibilidad_regreso_prision_no_responde = posibilidad_regreso_prision["P10_7"] == 9

In [192]:
posibilidad_regreso_prision[~ posibilidad_regreso_prision_no_sabe & ~ posibilidad_regreso_prision_no_responde].describe()

ID_PER         P10_7
count  11621.000000  11621.000000
mean    1653.919082      3.927975
std      892.547830      0.351627
min      101.143200      1.000000
25%      890.630600      4.000000
50%     1645.474000      4.000000
75%     2426.758300      4.000000
max     3203.333600      4.000000

In [193]:
posibilidad_regreso_prision = posibilidad_regreso_prision[~ posibilidad_regreso_prision_no_sabe & ~ posibilidad_regreso_prision_no_responde].groupby(["P10_7"]).count()

In [194]:
posibilidad_regreso = ["A. muy probable", "B. algo probable", "C. algo improbable", "D. muy improbable"]

In [195]:
posibilidad_regreso_prision["respuesta"] = posibilidad_regreso

In [196]:
primera_columna = posibilidad_regreso_prision.pop("respuesta")
posibilidad_regreso_prision.insert(0, "respuesta", primera_columna)
posibilidad_regreso_prision

respuesta  ID_PER
P10_7                            
1         A. muy probable      64
2        B. algo probable     138
3      C. algo improbable     369
4       D. muy improbable   11050

In [197]:
grafico_posibilidad_regreso_prision = alt.Chart(posibilidad_regreso_prision).mark_bar().encode(
    alt.X("respuesta", axis = alt.Axis(title = " ", labelAngle = 360, grid = False, labelFontSize = 12)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False, labelFontSize = 12)),
    alt.Color("respuesta", scale = alt.Scale(domain = ["A. muy probable", "B. algo probable", "C. algo improbable", "D. muy improbable"], range = ["#F0F0F0", "#FFEFB4", "#FFCD19", "#DBBF0D"])),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")]
).properties(
    title = "¿Qué tan probable considera regresar a prisión?",
    width = 450, height = 450,
).configure_title( 
  offset = 15, fontSize = 14,
)

In [198]:
grafico_posibilidad_regreso_prision

alt.Chart(...)

Conocimiento sobre programa postpenitenciario 
1. Persona == **ID_PER**
2. Pregunta == ¿Conoce o ha escuchado sobre algún programa o ayuda (programa postpenitenciario) para que los internos que salen libres se reincorporen a la vida en libertad (apoyo psicológico, encontrar trabajo, seguir estudiando, desarrollar algún deporte, etc.)? == **P10_8**
* **1** == Sí 
* **2** == No
* **8** == No sabe 
* **9** == No responde 


In [199]:
conoce_programa_postpenitenciario = enpol_expectativas_mujeres[["ID_PER", "P10_8"]]

In [200]:
conoce_programa_no_sabe = conoce_programa_postpenitenciario["P10_8"] == 8

In [201]:
conoce_programa_no_responde = conoce_programa_postpenitenciario["P10_8"] == 9

In [202]:
conoce_programa_postpenitenciario = conoce_programa_postpenitenciario[~ conoce_programa_no_sabe & ~ conoce_programa_no_responde].groupby(["P10_8"]).count()

In [203]:
conoce_programa = ["Si", "No"]

In [204]:
conoce_programa_postpenitenciario["respuesta"] = conoce_programa

In [205]:
primera_columna = conoce_programa_postpenitenciario.pop("respuesta")
conoce_programa_postpenitenciario.insert(0, "respuesta", primera_columna)
conoce_programa_postpenitenciario

respuesta  ID_PER
P10_8                  
1            Si    3291
2            No    8316

In [206]:
grafico_conoce_programa_postpenitenciario = alt.Chart(conoce_programa_postpenitenciario).mark_bar().encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False, labelFontSize = 12)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False, labelFontSize = 12)),
    alt.Color("respuesta", scale = alt.Scale(domain = ["Si", "No"], range = ["#FFCD19", "#8267BE"])),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")]
).properties(
    title = "¿Conoce algún programa postpenitenciario para reintegrarse a la vida en libertad?",
    width = 150, height = 400,
)

In [207]:
grafico_conoce_programa_postpenitenciario

alt.Chart(...)

1. Persona == **ID_PER**
2. Nombre de la Entidad Federativa == **NOM_ENT**
3. Pregunta == ¿En dónde se encuentran las personas que conocen sobre programas postpenitenciarios? Se emplearan los datos de la pregunta **P10_8**

In [208]:
entidades_conocimiento_programa = enpol_expectativas_mujeres[["ID_PER", "NOM_ENT", "P10_8"]]

In [209]:
conocimiento_programa = entidades_conocimiento_programa["P10_8"] == 1

In [210]:
entidades_conocimiento_programa[conocimiento_programa].groupby("NOM_ENT").count().sort_values(["ID_PER"])

ID_PER  P10_8
NOM_ENT                                       
Campeche                             11     11
Baja California Sur                  13     13
Colima                               14     14
Tlaxcala                             15     15
Zacatecas                            21     21
Quintana Roo                         25     25
Aguascalientes                       27     27
Yucatán                              28     28
Coahuila de Zaragoza                 29     29
Oaxaca                               33     33
Sinaloa                              37     37
Tabasco                              39     39
San Luis Potosí                      42     42
Guanajuato                           44     44
Chiapas                              45     45
Michoacán de Ocampo                  47     47
Veracruz de Ignacio de la Llave      50     50
Nayarit                              50     50
Guerrero                             52     52
Hidalgo                              58     58
Querétaro                            59     59
Tamaulipas                           74     74
Durango                              90     90
Sonora                              116    116
Puebla                              116    116
Nuevo León                          178    178
Jalisco                             194    194
Chihuahua                           205    205
Morelos                             237    237
Baja California                     283    283
Estado de México                    483    483
Ciudad de México                    576    576

1. Persona == **ID_PER**
2. Pregunta == Si tuviera la posibilidad, ¿usted participaría en estos programas o se beneficiaría de esa ayuda? == **P10_9**
* **1** == Sí 
* **2** == No
* **8** == No sabe 
* **9** == No responde 
* **NaN** == No aplica 

In [211]:
participacion_programa_postpenitenciario = enpol_expectativas_mujeres[["ID_PER", "P10_9"]].dropna()

In [212]:
participacion_programa_no_sabe = participacion_programa_postpenitenciario["P10_9"] == 8

In [213]:
participacion_programa_no_responde = participacion_programa_postpenitenciario["P10_9"] == 9

In [214]:
participacion_programa_postpenitenciario = participacion_programa_postpenitenciario[~ participacion_programa_no_sabe & ~ participacion_programa_no_responde].groupby(["P10_9"]).count()

In [215]:
participacion_programa = ["Si", "No"]

In [216]:
participacion_programa_postpenitenciario["respuesta"] = participacion_programa

In [217]:
primera_columna = participacion_programa_postpenitenciario.pop("respuesta")
participacion_programa_postpenitenciario.insert(0, "respuesta", primera_columna)
participacion_programa_postpenitenciario

respuesta  ID_PER
P10_9                  
1.0          Si    3052
2.0          No     221

In [218]:
grafico_participacion_programa_postpenitenciario = alt.Chart(participacion_programa_postpenitenciario).mark_bar().encode(
    alt.X("respuesta:N", axis = alt.Axis(title = " ", labelAngle = 360, grid = False, labelFontSize = 12)),
    alt.Y("ID_PER", axis = alt.Axis(title = " ", format = "1s", grid = False, labelFontSize = 12)),
    alt.Color("respuesta", scale = alt.Scale(domain = ["Si", "No"], range = ["#FFCD19", "#8267BE"])),
    tooltip = [alt.Tooltip("ID_PER:Q", title = "Número de informantes")]
).properties(
    title = "¿Participaría en un programa postpenitenciario?",
    width = 150, height = 400,
)

In [219]:
programa_postpenitenciario = (grafico_conoce_programa_postpenitenciario | grafico_participacion_programa_postpenitenciario).configure_title(
  anchor = "start", 
  offset = 15     
)

In [220]:
programa_postpenitenciario

alt.HConcatChart(...)

Los datos de este análisis se obtuvieron del Portal Oficial del Intituto Nacional de Geografía y Estadística:
* Instituto Nacional de Estadística y Geografía. "Encuesta Nacional De Población Privada De La Libertad (ENPOL) 2021". Inegi.Org.Mx, 2022. https://www.inegi.org.mx/programas/enpol/2021/#Microdatos.
<ul>


Las conclusiones de este análisis se puede consultar en la carpeta "Reportes" de este proyecto en un archivo PDF.
